addons/sale/models/account_move.py

```diff
@@ -29,14 +29,16 @@ def unlink(self):
             downpayment_lines.unlink()
         return res
 
-    @api.depends('invoice_user_id')
+    @api.depends('invoice_user_id', 'partner_id')
     def _compute_team_id(self):
         for ((user_id, company_id), moves) in groupby(
             self,
             key=lambda m: (m.invoice_user_id.id, m.company_id.id)
         ):
+            default_team_id = self.env.context.get('default_team_id', False) or moves[0].partner_id.team_id.id or moves[0].team_id.id
```

Comment:
this is wrong, you'll cannot decide the team of a group of invoices based on the values of the first one of the group, unless you make sure their configuration is the same.

Reply:
You are right, to solve this I changed the way the team_id is computed based on the method used for [sale.orders](https://github.com/odoo/odoo/blob/8dea99eed11c85644e5c24ff1805fc3c791afc7b/addons/sale/models/sale_order.py#L416)

Reply:
Well, the whole point of the groupby is to avoid timeouts and performance issues when sale is installed on top of an existing database, with a lot of existing invoices (where the field will be computed), or when a lot of invoices are created at once.
It should be kept somehow to avoid reintroducing the performance issue :thinking: